state/query/query.go

```diff
@@ -106,6 +106,14 @@ func (q *Query) UnmarshalJSON(data []byte) error {
 		if err != nil {
 			return err
 		}
+
+		jdata, err := json.Marshal(elem)
```

Comment:
Please use `github.com/mitchellh/mapstructure.Decode` rather than a json Marshal/Unmarshal since this is a fairly hot path.

Reply:
Tbh I think it could be better than that, I can simply unmarshal as json and right after build filters, I don’t know why it isn’t the actual approach, will try it out